### ISSUE TITLE
Article: Troubleshooting Fluentd

### ISSUE THREAD

**Issue by jonhandlermac** [fluent/fluentd-docs#380]
https://docs.fluentd.org/v0.12/articles/trouble-shooting

Turn on verbose logging says to use either TD_AGENT_ARGS or DAEMON_ARGS. I tried both and they threw an error. I set TD_AGENT_OPTIONS="... -vv" and that seems to have worked.

Jon

**Comment by fujimotos** [comment]
Thanks for reporting.

> Turn on verbose logging says to use either TD_AGENT_ARGS or DAEMON_ARGS. I tried both and they threw an error.

I can confirm this with td-agent v2.3.6: Setting `-vv` to `TD_AGENT_ARGS` or `DAEMON_ARGS` produces invalid command-line arguments and breaks the init.d script.

Evidently, the init.d script was cleaned up in td-agent v2.2.1, and since this version, the manual page became barely applicable (at least, in a straight-forward way).

### How to solve the issue

I think the solution is (as @jonhandlermac  suggests) to migrate to the newly introduced option `TD_AGENT_OPTIONS` as follows:

```bash
# Add this line to /etc/default/td-agent (in Debian/Ubuntu) or
# /etc/sysconfig/td-agent (in CentOS/RedHat)

TD_AGENT_OPTIONS='-vv'
```

This configuration should work for td-agent v2.2.1 or later.

Also, the commit treasure-data/omnibus-td-agent@71650be states that `TD_AGENT_OPTIONS` deprecates `DAEMON_ARGS`. So I believe the setting above is now the preferable way to set a verbose flag.

**Comment by fujimotos** [comment]
As #398 merged,  the v0.12 manual on the debugging flag is now up-to-date.
So mark this issue as resolved.

**Comment by fujimotos** [comment]
**Note:** the instruction above is only applicable to td-agent2.

Since newer installations use systemd (not sysvinit) to manage the td-agent daemon,
we need first implement an equivalent mechanism in td-agent.service first.

This will require some modifications in [the upstream project](https://github.com/treasure-data/omnibus-td-agent).

I'll work on this next.
